manifests/dotfile.pp

```diff
@@ -1,47 +1,34 @@
-# == Define: python::dotfile
 #
-# Manages any python dotfiles with a simple config hash.
-#
-# === Parameters
-#
-# [*ensure*]
-#  present|absent. Default: present
-#
-# [*filename*]
-#  Filename. Default: $title
-#
-# [*mode*]
-#  File mode. Default: 0644
-#
-# [*owner*]
-# [*group*]
-#  Owner/group. Default: `root`/`root`
-#
-# [*config*]
-#  Config hash. This will be expanded to an ini-file. Default: {}
-#
-# === Examples
-#
-# python::dotfile { '/var/lib/jenkins/.pip/pip.conf':
-#   ensure => present,
-#   owner  => 'jenkins',
-#   group  => 'jenkins',
-#   config => {
-#     'global' => {
-#       'index-url       => 'https://mypypi.acme.com/simple/'
-#       'extra-index-url => https://pypi.risedev.at/simple/
+# @summary Manages any python dotfiles with a simple config hash.
+#
+# @param ensure
+# @param filename Filename.
+# @param mode File mode.
+# @param owner user owner of dotfile
+# @param group group owner of dotfile
+# @param config Config hash. This will be expanded to an ini-file.
+#
+# @example Create a pip config in /var/lib/jenkins/.pip/
+#   python::dotfile { '/var/lib/jenkins/.pip/pip.conf':
+#     ensure => present,
+#     owner  => 'jenkins',
+#     group  => 'jenkins',
+#     config => {
+#       'global' => {
+#         'index-url'       => 'https://mypypi.acme.com/simple/'
+#         'extra-index-url' => 'https://pypi.risedev.at/simple/'
+#       }
 #     }
 #   }
-# }
 #
 #
 define python::dotfile (
-  $ensure   = 'present',
-  $filename = $title,
-  $owner    = 'root',
-  $group    = 'root',
-  $mode     = '0644',
-  $config   = {},
+  Enum['absent', 'present'] $ensure   = 'present',
+  $filename                           = $title,
+  String $owner                       = 'root',
+  String $group                       = 'root',
+  String $mode                        = '0644',
```

Comment:
Can you use the filemode type? https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/filemode.pp#L2